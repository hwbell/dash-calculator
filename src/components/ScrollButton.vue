<template>
  <div class="scroll-button-component">
    <a v-bind:href="target">
      <button class="btn btn-sm btn-light">
        {{ text }}
      </button>
    </a>
  </div>
</template>

<script>


export default {
  name: "ScrollButton",
  props: {
    target: String,
    text: String,
  },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
.scroll-button-component {
  margin: 4px;
}
</style>
