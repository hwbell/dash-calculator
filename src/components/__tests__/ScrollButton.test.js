import { mount } from '@vue/test-utils'
import ScrollButton from '../ScrollButton.vue'

const message = 'This is a scroll button component for us to start with!';

describe('ScrollButton', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = mount(ScrollButton, {
      props: {
        text: 'calculator',
        target: '#calculator'
      }
    })
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  
  test('renders a button for each object in the array map', () => {
    let messageDiv = wrapper.findAll('.scroll-button-component');
    expect(messageDiv.length).toBe(1);
    // expect(messageDiv.text()).toBe('This is a base component for us to start with!')
  })

  // test('receives and uses the props', () => {
    // e
  // })
})