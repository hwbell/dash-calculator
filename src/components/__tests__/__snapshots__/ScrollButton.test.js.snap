// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollButton renders correctly 1`] = `
<div
  class="scroll-button-component"
>
  <a>
    <button
      class="btn btn-sm btn-light"
    >
      
      
    
    </button>
  </a>
</div>
`;
